packages/arb-token-bridge-ui/src/util/networks.ts

```diff
@@ -190,3 +206,116 @@ export function getNetworkName(chainId: number) {
       return 'Unknown'
   }
 }
+
+export function getNetworkLogo(chainId: number) {
+  switch (chainId) {
+    // L1 networks
+    case ChainId.Mainnet:
+      return EthereumLogo
+
+    case ChainId.Rinkeby:
+      return EthereumLogo
+
+    case ChainId.Goerli:
+      return EthereumLogo
+
+    // L2 networks
+    case ChainId.ArbitrumOne:
+      return ArbitrumOneLogo
+
+    case ChainId.ArbitrumNova:
+      return ArbitrumNovaLogo
+
+    case ChainId.ArbitrumRinkeby:
+      return ArbitrumOneLogo
+
+    case ChainId.ArbitrumGoerli:
+      return ArbitrumOneLogo
+
+    default:
+      return EthereumLogo
+  }
+}
+
+type ExtendedWeb3Provider = Web3Provider & {
+  isMetaMask?: boolean
+  isImToken?: boolean
+}
+
+export const getSupportedNetworks = (chainId?: number) => {
+  // The list which will be available for network selection in navbar-dropdowns and unsupported-network-content
+  // If there is a chainId selected, it detects if it's a testnet and shows only those options
+
+  const mainNetworks = [
+    ChainId.ArbitrumOne,
+    ChainId.ArbitrumNova,
+    ChainId.Mainnet
+  ]
+  const testNetworks = [ChainId.ArbitrumGoerli, ChainId.Goerli]
+  const isTestnet = chainId ? isNetwork(chainId).isTestnet : false
+  return isTestnet ? testNetworks : mainNetworks
+}
+
+export type SwitchChainProps = {
+  chainId: number
+  provider: ExtendedWeb3Provider
+  onSuccess?: () => void
+  onError?: (err?: Error) => void
+  onSwitchChainNotSupported?: () => void
+}
+
+const isSwitchChainSupported = (provider: ExtendedWeb3Provider) => {
+  return provider?.provider?.isMetaMask || provider?.isImToken
+}
+
+const noop = () => {}
+
+export async function switchChain({
+  chainId,
+  provider,
+  onSuccess = noop,
+  onError = noop,
+  onSwitchChainNotSupported = noop
+}: SwitchChainProps) {
+  // do an early return if switching-chains is not supported by provider
+  if (!isSwitchChainSupported(provider)) {
+    onSwitchChainNotSupported?.()
+    return
+  }
+
+  // if all the above conditions are satisfied go ahead and switch the network
+  const hexChainId = hexValue(BigNumber.from(chainId))
+  const networkName = getNetworkName(chainId)
+
+  try {
+    await provider.send('wallet_switchEthereumChain', [
+      {
+        chainId: hexChainId
+      }
+    ])
+
+    onSuccess?.()
+  } catch (err: any) {
+    if (err.code === 4902) {
+      // https://docs.metamask.io/guide/rpc-api.html#usage-with-wallet-switchethereumchain
+      // This error code indicates that the chain has not been added to MetaMask.
+      await provider.send('wallet_addEthereumChain', [
+        {
+          chainId: hexChainId,
+          chainName: networkName,
+          nativeCurrency: {
+            name: 'Ether',
+            symbol: 'ETH',
+            decimals: 18
+          },
+          rpcUrls: [rpcURLs[chainId]],
+          blockExplorerUrls: [getExplorerUrl(chainId)]
+        }
+      ])
+
+      onSuccess?.()
+    } else {
+      onError?.(err)
```

Comment:
Should we keep `throw err` as on `App.tsx` previous version's L309?
then Sentry can catch it

Reply:
We can also explicitly say:

```
Sentry.captureException(err)
```